api/envoy/extensions/access_loggers/fluentd/v3/fluentd.proto

```diff
@@ -22,8 +24,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // the Fluentd Forward Protocol as described in: `Fluentd Forward Protocol Specification
 // <https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1>`_.
 // [#extension: envoy.access_loggers.fluentd]
-// [#next-free-field: 7]
+// [#next-free-field: 8]
 message FluentdAccessLogConfig {
+  message RetryOptions {
+    // The number of times the logger will attempt to connect to the upstream during reconnects.
+    // By default, there is no limit. The logger will attempt to reconnect to the upstream each time
+    // connecting to the upstream failed or the upstream connection had been closed for any reason.
+    google.protobuf.UInt32Value max_connect_attempts = 1;
```

Comment:
Sorry for being dense but I still do not understand why anyone would ever not want to reconnect. What is the use case for that?

/wait-any

Reply:
For not spinning operations that are likely to fail if the upstream is still unavailable after many reconnects. I understand that this may not be significant anyways, but the default now is there isn't a limit

Reply:
Let's just remove this option until someone asks for it. I just can't think of any case in which someone would want to use this. Trying to connect every 5s or so by default is not going to overburden anything.

/wait

Reply:
I myself will be using this option, therefore I was intetested in adding it :) I'd like to use this to avoid client port exhaustation. In some cases and during peak there's already high utilization of outbound connections, and to me it makes sense to reduce usage of any unnecesary resource if there a high probability the upstream is unavailable. Assuming there are multiple access loggers that use Fluentd and few dozens of cores (the Fluentd TCP client is per thread local), this may reach several hunder of ports, even if they are only being utilized for very short period of time.

Pls lmk if we can keep this one afterall, otherwise I'll delete

Reply:
Sure fair enough if you actually want to use it it can stay.